hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/HddsVolumeUtil.java

```diff
@@ -71,167 +61,65 @@ public static String getHddsRoot(String volumeRoot) {
-  }
-
-  /**
-   * Returns creationTime if it is valid. Throws an exception otherwise.
+   * Initialize db instance, rocksdb will load the existing instance
+   * if present and format a new one if not.
+   * @param containerDBPath
+   * @param conf
+   * @throws IOException
    */
-  @VisibleForTesting
-  public static long getCreationTime(Properties props, File versionFile)
-      throws InconsistentStorageStateException {
-    String cTimeStr = getProperty(props, OzoneConsts.CTIME, versionFile);
-
-    long cTime = Long.parseLong(cTimeStr);
-    long currentTime = Time.now();
-    if (cTime > currentTime || cTime < 0) {
-      throw new InconsistentStorageStateException("Invalid Creation time in " +
-          "Version File : " + versionFile + " - " + cTime + ". Current system" +
-          " time is " + currentTime);
-    }
-    return cTime;
+  public static void initPerDiskDBStore(String containerDBPath,
+      ConfigurationSource conf) throws IOException {
+    DatanodeStore store = BlockUtils.getUncachedDatanodeStore(containerDBPath,
+        OzoneConsts.SCHEMA_V3, conf, false);
+    BlockUtils.addDB(store, containerDBPath, conf, OzoneConsts.SCHEMA_V3);
   }
 
   /**
-   * Returns layOutVersion if it is valid. Throws an exception otherwise.
+   * Load already formatted db instances for all HddsVolumes.
+   * @param hddsVolumeSet
+   * @param dbVolumeSet
+   * @param logger
    */
-  @VisibleForTesting
-  public static int getLayOutVersion(Properties props, File versionFile) throws
-      InconsistentStorageStateException {
-    String lvStr = getProperty(props, OzoneConsts.LAYOUTVERSION, versionFile);
-
-    int lv = Integer.parseInt(lvStr);
-    if (HDDSVolumeLayoutVersion.getLatestVersion().getVersion() != lv) {
-      throw new InconsistentStorageStateException("Invalid layOutVersion. " +
-          "Version file has layOutVersion as " + lv + " and latest Datanode " +
-          "layOutVersion is " +
-          HDDSVolumeLayoutVersion.getLatestVersion().getVersion());
-    }
-    return lv;
-  }
-
-  public static String getProperty(
-      Properties props, String propName, File
-      versionFile
-  )
-      throws InconsistentStorageStateException {
-    String value = props.getProperty(propName);
-    if (StringUtils.isBlank(value)) {
-      throw new InconsistentStorageStateException("Invalid " + propName +
-          ". Version File : " + versionFile + " has null or empty " + propName);
+  public static void loadAllHddsVolumeDbStore(MutableVolumeSet hddsVolumeSet,
+      MutableVolumeSet dbVolumeSet, Logger logger) {
+    // Scan subdirs under the db volumes and build a one-to-one map
+    // between each HddsVolume -> DbVolume.
+    mapDbVolumesToDataVolumesIfNeeded(hddsVolumeSet, dbVolumeSet);
+
+    for (HddsVolume volume : StorageVolumeUtil.getHddsVolumesList(
+        hddsVolumeSet.getVolumesList())) {
+      try {
+        volume.loadDbStore();
+      } catch (IOException e) {
+        onFailure(volume);
```

Comment:
Shall we just fail the volume, like "hddsVolumeSet.failVolume(volume.getStorageDir().getPath());"?